src/main/java/org/opensearch/knn/common/launchcontrol/LaunchControlUtils.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.common.launchcontrol;
+
+import org.apache.lucene.search.Query;
+import org.opensearch.knn.index.query.KNNQuery;
+import org.opensearch.knn.index.query.nativelib.NativeEngineKnnVectorQuery;
+
+public final class LaunchControlUtils {
+
+    public static Query getNativeKNNQuery(final KNNQuery knnQuery) {
```

Comment:
This function can easily be part of KNNQueryFactory.java and `KNNLaunchControl.isFeatureEnabled(KNNLaunchControl.KNN_LAUNCH_QUERY_REWRITE_ENABLED_SETTING.getKey())` can be added as isNativeQueryRewriteEnabled()

Reply:
Separate should work here especially if we end up making changes for radial query

Reply:
Even in that case isn't KNNQueryBuilder going to use this flag only. This being a public interface and just being used by KNNQueryFactory doesn't seem correct. and lets say Radial search wants to use, then it should use isRewriteFeatureEnabled and doesn't need to know if query conversion is needed or not.

Reply:
> Even in that case isn't KNNQueryBuilder going to use this flag only. This being a public interface and just being used by KNNQueryFactory doesn't seem correct.

Adding it in querybuilder will cause cyclic dependency. Querybuilder depends on queryfactory which will depend on a method in query builder

> This being a public interface and just being used by KNNQueryFactory doesn't seem correct
> then it should use isRewriteFeatureEnabled and doesn't need to know if query conversion is needed or not.

The idea is to use a common method to decide if native engine query or not irrespective what code base accesses it. the definition is tight enough to not accept any other query. There isn't a good place, isolating seems like a right approach. I will change the name of the class to narrow down its responsibility